### PR TITLE
*: update binlog for new mechanism that save to TiDB local file.

### PR DESCRIPTION
**Pull request by coocood**
@ngaut @siddontang @shenli PTAL

### CODE REVIEW

File: proto/binlog/binlog.proto [hunk 1]
```diff
@@ -10,16 +10,20 @@ option (gogoproto.unmarshaler_all) = true;
...
-    optional int64 table_id     = 1 [(gogoproto.nullable) = false];
-    repeated int64 inserted_ids = 2;
-    repeated int64 updated_ids  = 3;
-    repeated int64 deleted_ids  = 4;
+    optional int64 table_id      = 1 [(gogoproto.nullable) = false];
 
-    // For tables with non-interger types primary key.
-    repeated bytes deleted_pks = 5;
+    // For inserted rows and updated rows, we save all column values of the row.
+    repeated bytes inserted_rows = 2;
```
Comment (shenli) [inline proto/binlog/binlog.proto]: Why save row values?

Comment (coocood) [inline proto/binlog/binlog.proto]: It's going to be saved in local file, saving the row values so we don't need to look up the row in the TiKV.

**Comment by coocood** [comment]
@shenli @siddontang @iamxy PTAL

File: proto/binlog/binlog.proto [hunk 2]
```diff
@@ -10,21 +10,43 @@ option (gogoproto.unmarshaler_all) = true;
...
+    repeated int64 deleted_ids   = 4;
+
+    // If the table has PK but PK is not handle, we save the PK of the deleted row.
+    repeated bytes deleted_pks   = 5;
+
+    // If the table doesn't have PK, we save the row value of the deleted row.
+    repeated bytes deleted_rows  = 6;
+}
+
+enum BinlogType {
```
Comment (siddontang) [inline proto/binlog/binlog.proto]: do we need the type? @iamxy

**Comment by shenli** [comment]
LGTM
@siddontang @iamxy PTAL

File: proto/binlog/binlog.proto (hunk 2 above)
Comment (iamxy) [inline proto/binlog/binlog.proto]: Indeed, according to the design, when a transaction is committed, it should write two binlogs with different type.

**Comment by siddontang** [comment]
LGTM

**Comment by coocood** [comment]
Store raw bytes for prewrite mutations, binlog processor server need to store it in rocksDB, save raw bytes in Binlog type, so we don't need to unmarshal it and re-marshal it.

@iamxy @siddontang @shenli PTAL

**Comment by iamxy** [comment]
LGTM

**Pull request merged**